### PR TITLE
fix(review-bot): parse @handles anywhere on an r? line

### PR DESCRIPTION
**Pull request by Nils-Fedrigo**
## Description

The engbot review parser previously stopped collecting reviewers at the first non-mention token. That meant `r? @pmilliotte or @Nils-Fedrigo` only requested `@pmilliotte`.

This changes `parseReviewRequests` to collect every whitespace-delimited GitHub mention (and bare `cc`) anywhere after a column-zero `r?`, so prose like `or` / `and` / `please` no longer truncates the list.

Also updates the product contract + README, and adds `node:test` coverage for the motivating `or` case and related edge cases.

## Tests

- Added `.github/actions/review-bot/review-bot.test.ts`
- Ran `node --experimental-strip-types --test review-bot.test.ts` in `.github/actions/review-bot` (8/8 passing)

Demo for the Fabien-style request:

```text
input:  r? @pmilliotte or @Nils-Fedrigo
parsed: reviewers: ["pmilliotte", "nils-fedrigo"]
```

## Risk

Low. Behavior is additive for existing consecutive-mention requests. More reviewers may be requested when authors write prose between handles (intended). Team mentions and invalid tokens remain ignored.

## Deploy Plan

Merge to default branch so the review-bot action (loaded from default branch) picks up the parser change. No migration or config change.

[Slack Thread](https://dust4ai.slack.com/archives/C09GELMTTRT/p1789118058442359?thread_ts=1789118058.442359&cid=C09GELMTTRT)

<div><a href="https://cursor.com/agents/bc-a760e347-0e78-5c7b-8312-8468f90f723a?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-a760e347-0e78-5c7b-8312-8468f90f723a&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Review by spolu — COMMENTED** [review]
LGTM

File: .github/actions/review-bot/review-bot.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        },
+      ]
+    );
+  });
+
+  it("recognizes bare cc anywhere on the request line", () => {
+    assert.deepEqual(parseReviewRequests("r? @spolu please cc @flvndvd"), [
+      {
+        line: "r? @spolu please cc @flvndvd",
+        reviewers: ["spolu", "flvndvd"],
```
Comment (spolu) [inline .github/actions/review-bot/review-bot.test.ts:44]: mmmmm actually

if I cc @flvndvd I don't want him to be a reviewer do I?

Comment (Nils-Fedrigo) [reply .github/actions/review-bot/review-bot.test.ts:44]: I would personally! Or at least have him be pinged on the slack message
I can remove that if you prefer

Comment (spolu) [reply .github/actions/review-bot/review-bot.test.ts:44]: Let's give it a try!
